archive/frames/models.py

```diff
@@ -133,11 +133,13 @@ def get_header_dict(self):
             archive_settings.PUBLIC_DATE_KEY: self.public_date,
         }
 
-    def as_dict(self):
+    def as_dict(self, include_thumbnails=False):
         ret_dict = model_to_dict(self, exclude=('related_frames', 'area'))
         ret_dict['version_set'] = [v.as_dict() for v in self.version_set.all()]
-        ret_dict['url'] = self.url
-        ret_dict['filename'] = self.filename
+        ret_dict['url'] = self.url if self.version_set.exists() else None
+        ret_dict['filename'] = self.filename if self.version_set.exists() else None
+        if include_thumbnails:
```

Comment:
I couldn't figure out a nice way to do this in the FrameFilter, so I just check for this here coming in from the viewset